packages/client/docusaurus/docs/client/05-advanced/03-permissions-and-capabilities.mdx

```diff
@@ -3,4 +3,147 @@ id: 03-permissions-and-capabilities
+const canSendAudio = call.permissionsContext.hasPermission(
+  OwnCapability.SEND_AUDIO,
+);
+```
+
+<SDKSpecific name="react">
+In our React Video SDK, you can use the [`useHasPermissions`](../04-call-engine/hooks.md#useHasPermissions) hook to check for permissions.
+
+```tsx
+const canSendAudio = useHasPermissions(OwnCapability.SEND_AUDIO);
+```
+
+</SDKSpecific>
+
+### Request permissions
+
+Every user may request permissions to perform certain action depending on the call type and call settings.
+For example, in an audio-room call type, only the hosts have `send-audio` permission by default.
+Other users should request this permission before they can start sending audio if the call settings allow it.
+
+```ts
+import { OwnCapability } from '@stream-io/video-client';
+
+const call = streamVideoClient.call(type, id);
+if (!call.permissionsContext.canRequest(OwnCapability.SEND_AUDIO)) {
+  console.log('The host has disabled the ability to request this permission');
+  return;
+}
+await call.requestPermissions({
+  permissions: [OwnCapability.SEND_AUDIO],
+});
+```
+
+### Approving permission requests
+
+Call hosts and moderators can approve permission requests from other users.
+Whenever a user requests a permission, a `call.permission_request` event will be emitted on the `Call` instance.
+You can listen to this event and approve the request.
+
+```ts
+import { PermissionRequestEvent } from '@stream-io/video-client';
+
+const call = streamVideoClient.call(type, id);
+call.on('permission_request', async (request: PermissionRequestEvent) => {
+  if (shouldApproveRequest(request)) {
+    await call.updateUserPermissions({
+      user_id: request.user_id,
+      grant_permissions: [...request.permissions],
+    });
+  }
+});
+```
+
+### Moderation
+
+At any time, a moderator or host can decide to either grant or revoke certain permission to any participant.
+
+```ts
+import { OwnCapability } from '@stream-io/video-client';
+
+const call = streamVideoClient.call(type, id);
+await call.updateUserPermissions({
+  user_id: 'demo-user',
+  grant_permission: [OwnCapability.SEND_AUDIO, OwnCapability.SEND_VIDEO],
+  revoke_permissions: [OwnCapability.SCREENSHARE],
+});
+```
+
+The end user would get notified via a WebSocket event with a type: `call.permissions_updated`.
+In the case of revoked permissions, the SDK would automatically stop publishing the appropriate tracks.
+
+### Muting participants
+
+In addition to granting or revoking permissions, a moderator or host can also mute a participant.
+This is a common scenario as quite often, participants may be a source of unwanted noise or distraction.
+
+```ts
+const call = streamVideoClient.call(type, id);
+await call.muteUser('demo-user-id', 'audio');
+await call.muteUser('demo-user-id', 'video');
+await call.muteUser('demo-user-id', 'screenshare');
+
+// or, mute in bulk
+await call.muteUser(['demo-user-id', 'demo-user-id-2'], 'audio');
+
+// or, mute all
+await call.muteAllUsers('audio');
+```
```

Comment:
Should we add an example of how to mute yourself here? That should be a quite common scenario as well, right?

Reply:
Yes, essentially that would be using our own "user_id" as an argument. I'll add support for it in the Client.